pkg/machines/components/create-vm-dialog/pxe-helpers.js

```diff
@@ -0,0 +1,187 @@
+        if (ips[i].dhcp.bootp)
+            return true;
+
+    return false;
+}
+
+/**
+ * Returns the first available Network Resource to be used for showing to PXE Network Sources list.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXEInitialNetworkSource(nodeDevices, virtualNetworks, connectionName) {
+    if (virtualNetworks.length > 0)
+        return cockpit.format('network=$0', virtualNetworks[0].name);
+
+    let netNodeDevices = getNodeDevicesOfType(nodeDevices, connectionName, 'net');
+
+    /* Do not show to the user the libvirt virbrX-nic devices since these are
+     * supposed to be managed through Virtual Networks
+     */
+    let libvirtVirBridges = getLibvirtNetworkBridges(virtualNetworks);
+    netNodeDevices = netNodeDevices.filter(netNodeDevice => {
+        for (let i in libvirtVirBridges) {
+            if (netNodeDevice.capability.interface == (libvirtVirBridges[i] + '-nic'))
+                return false;
+        }
+        return true;
+    });
+
+    if (netNodeDevices.length > 0)
+        return cockpit.format('type=direct,source=$0', netNodeDevices[0].capability.interface);
+}
+
+/**
+ * Returns the Select Entries rows for the PXE Network Sources.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXENetworkRows(nodeDevices, virtualNetworks, connectionName) {
+    /* Do not show to the user the libvirt virbrX-nic devices since these are
+     * supposed to be managed through virtual networks
+     */
+    const libvirtVirBridges = getLibvirtNetworkBridges(virtualNetworks);
+    let netNodeDevices = getNodeDevicesOfType(nodeDevices, connectionName, 'net');
+
+    netNodeDevices = netNodeDevices.filter(netNodeDevice => {
+        for (let i in libvirtVirBridges) {
+            if (netNodeDevice.capability.interface == (libvirtVirBridges[i] + '-nic'))
+                return false;
+        }
+        return true;
+    });
+
+    let virtualNetworkRows = virtualNetworks.map(network =>
+        <Select.SelectEntry
+            data={cockpit.format('network=$0', network.name)}
+            key={cockpit.format('network=$0', network.name)}>
+            {cockpit.format("$0 $1: $2", _("Virtual Network"), network.name, getVirtualNetworkDescription(network))}
+        </Select.SelectEntry>
+    );
+
+    let netNodeDevicesRows = netNodeDevices.map(netNodeDevice => {
+        let iface = netNodeDevice.capability.interface;
+
+        return (
+            <Select.SelectEntry
+                data={cockpit.format('type=direct,source=$0', iface)}
+                key={cockpit.format('type=direct,source=$0', iface)}>
```

Comment:
Please move the `cockpit.format` into the assignment of `iface` to avoid doing it twice.

Reply:
done